### ISSUE TITLE
White screen error

### ISSUE THREAD

**Issue by lukemaster** [electron/electron#10535]
* Electron version: 1.8 beta
* Operating system: MacOS Sierra

### Expected behavior

All right.

### Actual behavior

I got a memory allocation error.

### How to reproduce

I was using electron normally, and I got this error in console.

It also occurs when you select the application while it was minimized, or in another desktop.

This is a spanshot of the console:

<img width="1440" alt="screen shot 2017-09-16 at 19 06 22" src="https://user-images.githubusercontent.com/1827014/30514377-d0514cf4-9b13-11e7-938d-3089fe18f233.png">

**Comment by missdora** [comment]
I have the same problem

**Comment by AlbearKamoo** [comment]
I'm running version 1.8 and I'm encountering the same issue with very similar crash logs. In my case it usually occurs when performing network requests.

**Comment by codebytere** [comment]
@AlbearKamoo @missdora do you have a sample file that reliably causes this crash? I can take a look at that and see what i can dig up

**Comment by AlbearKamoo** [comment]
I'm afraid I'm not able to release the files that are reliably causing the issue (private company code).

All I can provide for now is the following crash log. Also, we rolled back to 1.7.6, and although it was more stable, there were a still few instances where we could produce a crash with similar logs.

[1476:0925/115053.108995:FATAL:partition_alloc.cc(934)] Check failed: page->num_allocated_slots != -1. 
0   Electron Framework                  0x000000010f73da2c crashpad::CloseMultipleNowOrOnExec(int, int) + 969932
1   Electron Framework                  0x000000010f705353 crashpad::CloseMultipleNowOrOnExec(int, int) + 738803
2   Electron Framework                  0x000000010f71cdb1 crashpad::CloseMultipleNowOrOnExec(int, int) + 835665
3   Electron Framework                  0x00000001123b906c crashpad::CloseMultipleNowOrOnExec(int, int) + 47612684
4   libnode.dylib                       0x0000000114d6c5ce v8::internal::DeferredHandles::Iterate(v8::internal::ObjectVisitor*) + 20670
5   libnode.dylib                       0x0000000114d6c50e v8::internal::DeferredHandles::Iterate(v8::internal::ObjectVisitor*) + 20478
6   libnode.dylib                       0x00000001151f767a v8::internal::AllocationSpaceName(v8::internal::AllocationSpace) + 27370
7   libnode.dylib                       0x00000001151f0570 v8::internal::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, v8::internal::MachineType) + 76800
8   libnode.dylib                       0x00000001151f08e4 v8::internal::operator<<(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, v8::internal::MachineType) + 77684
9   libnode.dylib                       0x000000011520011c v8::internal::AllocationSpaceName(v8::internal::AllocationSpace) + 62860
10  Electron Framework                  0x000000010f77d0d2 crashpad::CloseMultipleNowOrOnExec(int, int) + 1229682
11  Electron Framework                  0x000000010f724757 crashpad::CloseMultipleNowOrOnExec(int, int) + 866807
12  libsystem_pthread.dylib             0x00007fffb881493b _pthread_body + 180
13  libsystem_pthread.dylib             0x00007fffb8814887 _pthread_body + 0
14  libsystem_pthread.dylib             0x00007fffb881408d thread_start + 13

**Comment by codebytere** [comment]
ok thanks! i'll see what i can do

**Comment by codebytere** [comment]
Previous instances of this error tell me that it's most likely an issue that crops up during garbage collection. I've found a series of issues that might shed light on a more specific cause:
https://github.com/electron/electron/issues/7732 https://github.com/electron/electron/issues/7732 https://github.com/electron/electron/issues/10455. Let me know if any of these help, but i think it might potentially be a native module issue and thus something we're unable to fix.

**Comment by AlbearKamoo** [comment]
We narrowed down the root cause of the issue and once we changed some things in our code it no longer crashes!

Apparently we were using some node libraries on the client-side Renderer process and, as you pointed out, they were triggering a garbage collection error that crashed the application. Specifically we were using the 'http' module for some of our API requests. Once we switched those to pure XHR requests the app no longer crashed.
